### PR TITLE
Dev/oo 724 adding zusatzabo in abo

### PR DESCRIPTION
**Pull request by mcmpp**

### CODE REVIEW

**Review by AlwinEgger — CHANGES_REQUESTED** [review]
Travis is complaining:
"[error] /home/travis/build/OpenOlitor/openolitor-server/src/main/scala/ch/openolitor/stammdaten/StammdatenUpdateService.scala:427: not found: value modifyKoerbeForAbo"

File: src/main/scala/ch/openolitor/stammdaten/StammdatenInsertService.scala [hunk 1]
```diff
@@ -473,6 +476,74 @@ class StammdatenInsertService(override val sysConfig: SystemConfig) extends Even
...
+
+  def defaultDateStart(date1: LocalDate, date2: Option[LocalDate]): LocalDate = {
+    date2 match {
+      case Some(d2) => {
+        if ((date1 compareTo d2) > 0) date1 else d2
+      }
+      case None => date1
+    }
+  }
+
```
Comment (fabianpage) [inline src/main/scala/ch/openolitor/stammdaten/StammdatenInsertService.scala]: Maybe something like that is easier to comprehend?
``` scala

def defaultDateStart2(date1: LocalDate, date2: Option[LocalDate]): LocalDate = {
  date2.map { d2 =>
    if ((date1 compareTo d2) > 0) date1 else d2
  }.getOrElse(date1)
}
```

Comment (toggm) [reply src/main/scala/ch/openolitor/stammdaten/StammdatenInsertService.scala]: Or even:
```
def defaultDateStart2(date1: LocalDate, date2: Option[LocalDate]): LocalDate = {
 (date1, date2) match {
      case (d1, None) => d1
      case (d1, Some(d2)) if (d1 compareTo d2) > 0 => d1
      case (_, d2) => d2
    }    
}
```

Then you see all the decisions in one level.

File: src/main/scala/ch/openolitor/stammdaten/StammdatenInsertService.scala [hunk 2]
```diff
@@ -473,6 +476,74 @@ class StammdatenInsertService(override val sysConfig: SystemConfig) extends Even
...
+          case Some(d2) => {
+            if ((d1 compareTo d2) < 0) date1 else date2
+          }
+          case None => date1
+        }
+      }
+      case None => date2
+    }
+  }
+
```
Comment (fabianpage) [inline src/main/scala/ch/openolitor/stammdaten/StammdatenInsertService.scala]: Maybe something like that is easier to comprehend?

``` scala 

def defaultDateEnd2(date1: Option[LocalDate], date2: Option[LocalDate]): Option[LocalDate] = {
  (date1, date2) match {
    case (Some(d1), Some(d2)) => if ((d1 compareTo d2) < 0) date1 else date2
    case (Some(d1), _) => date1
    case _ => date2
  }
}
```

File: src/main/scala/ch/openolitor/stammdaten/StammdatenInsertService.scala (hunk 1 above)
Comment (mcmpp) [reply src/main/scala/ch/openolitor/stammdaten/StammdatenInsertService.scala]: I like yours Mike, I'll change it !

**Review by AlwinEgger — APPROVED** [review]

**Pull request merged**